src/App.jsx

```diff
@@ -14,33 +15,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getPrepareGoods(goods, { sortByAlphabet, reverse, sortByLength }) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortByAlphabet) {
+    preparedGoods.sort((a, b) => a.localeCompare(b));
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortByLength) {
+    preparedGoods.sort((a, b) => a.length - b.length);
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortByAlphabet, setSortByAlphabet] = useState(false);
+  const [sortByLength, setSortByLength] = useState(false);
```

Comment:
You can sort either alphabetically or by length, but it can't be both, so it's better to combine these two states and set the initial value to an empty string, and then click the button to write one of the two values to this state